common/service/config/config.go

```diff
@@ -329,9 +329,14 @@ type (
 	}
 
 	// PublicClient is config for connecting to cadence frontend
+	// Currently supporting single IP address(HostPort) and DNS and Port. Only one of them is required.
 	PublicClient struct {
 		// HostPort is the host port to connect on
 		HostPort string `yaml:"hostPort" validate:"nonzero"`
```

Comment:
What happens if I don't configure hostPort? Will the validation fail?

Reply:
The new code also works with IP address, so I would keep using the same config of hostport and let it non-zero